dev-utilities/build-utils/list_dependencies.sh

```diff
@@ -16,4 +16,16 @@
 #  limitations under the License.
 #
 
-{ mvn dependency:list || { echo "ERROR:  Failed to run mvn dependency:list" ; exit 1 ; } ; mvn dependency:list -PHDP-2.5.0.0 || { echo "ERROR:  Failed to run mvn dependency:list -PHDP-2.5.0.0" ; exit 1 ; } ; } | grep "^\[INFO\]   " | awk '{print $2}' | grep -v "org.apache" | grep -v "test" | grep -v "provided" | grep -v "runtime" | grep -v ":system" |  sort | uniq
+DEPS=$(mvn dependency:list)
+rc=$?
```

Comment:
Is this just a refactor for readability?  Or was the functionality changed at all here?

Reply:
It's one of the main fixes, because of the incorrect failure handling described in the ticket. It has to do with exiting during pipes not stopping the chained call.